src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/controllers/UserAuthController.java

```diff
@@ -168,12 +172,24 @@ LoginResponse create(final HttpServletRequest request,
         if (person.getEmail() == null) {
           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "email_required");
         }
+
+        send_verification_email = true;
+        token = "";
+        PersonEmailVerification personEmailVerification = null;
+
+        try {
+          personEmailVerification = personEmailVerificationService.create(person,
+              request.getRemoteAddr(), request.getHeader("User-Agent"));
+        } catch (Exception e) {
+          personRepository.delete(person);
+          throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
+              "creating_email_verification_failed");
```

Comment:
I agree here. I'm not sure we should delete a user because of an exception. If we want to have some kind of boolean result that we use for determining if the verification failed, that's something to discuss. But potentially deleting a user due to an error in the code seems like a bad UX.